spec/ordinary-and-exotic-object-behaviours.html

```diff
@@ -0,0 +1,90 @@
+<!DOCTYPE html>
+<meta charset="utf8" />
+
+<emu-clause id="sec-ordinary-and-exotic-objects-behaviours">
+  <h1>Ordinary and Exotic Objects Behaviours</h1>
+
+  <emu-clause id="sec-built-in-exotic-object-internal-methods-and-slots">
+    <h1>Built-in Exotic Object Internal Methods and Slots</h1>
+    <emu-clause id="sec-record-exotic-objects">
+      <h1>Record Exotic Objects</h1>
+      <p>A Record object is an exotic object that encapsuates a Record value and exposes virtual string-indexed data properties corresponding to the individual keys set on the underlying Record value. All keys properties are non-writable and non-configurable.</p>
+
+      <p>An object is a <dfn id="record-exotic-object">Record exotic object</dfn> (or simply, a Record object) if its [[GetOwnPoperty]], [[DefineOwnProperty]], and [[OwnPropertyKeys]] internal methods.</p>
+
+      <p>Record exotic objects have the same internal slots as ordinary objects. They also have a [[RecordData]] internal slot.</p>
+
+
+      <emu-clause id="sec-record-exotic-objects-getownproperty-p">
+        <h1>[[GetOwnProperty]] ( _P_ )</h1>
+        <p>When the [[GetOwnProperty]] internal method of a Record exotic object _R_ is called with property key _P_, the following steps are taken:</p>
+        <emu-alg>
+          1. Assert: IsPropertyKey(_P_) is *true*.
+          1. If Type(_P_) is a Symbol, throw a *TypeError* exception.
+          1. Return ! RecordGetOwnProperty(_R_, _P_).
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-record-exotic-objects-defineownproperty-p-desc">
+        <h1>[[DefineOwnProperty]] ( _P_, _Desc_ )</h1>
+        <p>When the [[DefineOwnProperty]] internal method of a Record exotic object _R_ is called with property key _P_, and Property Descriptor _Desc_ the following steps are taken:</p>
+        <emu-alg>
+          1. Assert: IsPropertyKey(_P_) is *true*.
+          1. Throw a *TypeError* exception.
```

Comment:
This needs to accept property definitions that don't change the value. See module namespace exotic objects for an example. The same goes for setting the prototype. https://tc39.es/ecma262/#sec-module-namespace-exotic-objects